pkg/asset/cluster/azure/azure.go

```diff
@@ -12,3 +21,44 @@ func Metadata(config *types.InstallConfig) *azure.Metadata {
 		Region: config.Platform.Azure.Region,
 	}
 }
+
+// PreTerraform performs any infrastructure initialization which must
+// happen before Terraform creates the remaining infrastructure.
+func PreTerraform(ctx context.Context, clusterID *installconfig.ClusterID, installConfig *installconfig.InstallConfig) error {
+	clusterName := installConfig.Config.ObjectMeta.Name
+	azureDNS, _ := azureconfig.NewDNSConfig()
+	record := fmt.Sprintf("api.%s", clusterName)
+
+	exists := false
```

Comment:
nit: exists -> found, i think that conveys better meaning here..

Reply:
Fixed.